components/emoji_picker/emoji_picker.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div className="d-emoji-picker">
+    <div className="d-emoji-picker--header">
+      <emojiTabset
+        :recently-used-tab="showRecentlyUsedTab"
+        @tabset-selected="scrollToSelectedTabset"
+      />
+    </div>
+    <div className="d-emoji-picker--body">
+      <emojiSearch v-model="searchQuery" />
+      <emojiSelector
+        :emoji-filter="searchQuery"
+        :skin-tone="skinTone"
+        @emoji-data="updateEmojiData"
+        @selected-emoji="emits('selected-emoji', $event)"
+      />
+    </div>
+    <div className="d-emoji-picker--footer">
+      <emojiData :emoji-data="emojiData" />
+      <emojiSkinSelector @skin-tone="updateSkinTone" />
+    </div>
+  </div>
+</template>
+
+<script setup>
+import { computed, onBeforeUnmount, ref } from 'vue';
+
+const props = defineProps({
+  /**
+   * The list of recently used emojis
+   * This list is necessary to fill the recently used tab
+   * @type {Array}
+   * @default []
+   * @example
+   * <dt-emoji-picker :recentlyUsedEmojis="['😀', '😁']" />
+   */
+  recentlyUsedEmojis: {
+    type: Array,
+    default: () => [],
+  },
```

Comment:
ye... still dunno.. will figure out later lol, Im kind of sure this prop will be needed, but honestly dont fully convince how it will work, but it will sth like you are saying

Reply:
Since this stub serves as a tech spec we should probably figure out and write some details about how we are going to do this. A primary purpose of this exercise is that we can determine if an idea is going to work or not before we start actually coding it. This prevents us from having to go back and rewrite things later.

Reply:
I cannot see the big deal here, maybe thats the problem..., what I expect to receive its the emoji object data similiar as what we have currently.

<img width="488" alt="image" src="https://user-images.githubusercontent.com/89984179/226632902-201f4c68-c8a9-4659-8436-4139c1277614.png">

Reply:
I'd say it's unlikely the consumer will pass in an entire emoji data object into here. Probably just an array of unicode characters?

Reply:
Mhmm.. it is possible, its depends on how I will manage show the emojis.
I will put a TODO here to try to simplify this.